sdk/eventhub/Azure.Messaging.EventHubs/src/Producer/EventHubBufferedProducerClientOptions.cs

```diff
@@ -16,57 +16,127 @@ namespace Azure.Messaging.EventHubs.Producer
+        ///</summary>
+        ///
+        /// <value>
+        ///   By default, idempotent retries are disabled.
+        /// </value>
+        ///
+        /// <remarks>
+        ///   It is important to note that enabling idempotent retries does not guarantee exactly-once semantics.  The existing
+        ///   Event Hubs at-least-once delivery contract still applies and event duplication is unlikely, but possible.
+        /// </remarks>
+        ///
+        public bool EnableIdempotentRetries { get; set; }
+
+        /// <summary>
+        ///   The amount of time to wait for a new event to be enqueued in the buffer before publishing
+        ///   a partially full batch.
         /// </summary>
         ///
         /// <value>
         ///   The default wait time is 250 milliseconds.
+        ///
+        ///   <para>If <c>null</c>, attempts to enqueue events will wait forever for room in the buffer to become available
+        ///   unless the operation is canceled.</para>
         /// </value>
         ///
-        public TimeSpan? MaximumWaitTime { get; set; } = TimeSpan.FromMilliseconds(250);
+        /// <exception cref="ArgumentOutOfRangeException">Occurs when the requested wait time is negative.</exception>
+        ///
+        public TimeSpan? MaximumWaitTime
+        {
+            get => _maximumWaitTime;
+
+            set
+            {
+                if (value.HasValue)
+                {
+                    Argument.AssertNotNegative(value.Value, nameof(MaximumWaitTime));
+                }
+
+                _maximumWaitTime = value;
+            }
+        }
 
         /// <summary>
-        ///   The total number of events that can be buffered for publishing at a given time across all partitions.
-        ///   Once this capacity is reached, more events can enqueued by calling <see cref="EventHubBufferedProducerClient.EnqueueEventAsync(EventData, EnqueueEventOptions, System.Threading.CancellationToken)" /> or
-        ///   <see cref="EventHubBufferedProducerClient.EnqueueEventsAsync(System.Collections.Generic.IEnumerable{EventData}, EnqueueEventOptions, System.Threading.CancellationToken)" />, which will automatically wait for room to be available.
+        ///   The total number of events that can be buffered for publishing at a given time for a given partition.  Once this capacity is reached, more events can enqueued by calling
+        ///   <see cref="EventHubBufferedProducerClient.EnqueueEventAsync(EventData, EnqueueEventOptions, System.Threading.CancellationToken)" /> or
+        ///   <see cref="EventHubBufferedProducerClient.EnqueueEventsAsync(IEnumerable{EventData}, EnqueueEventOptions, System.Threading.CancellationToken)" />, which will automatically wait for room to be available.
         /// </summary>
         ///
         /// <value>
-        ///   The default limit is 2,500 queued events.
+        ///   The default limit is 1,500 queued events for each partition.
```

Comment:
nit: the comma is culture specific.
```suggestion
        ///   The default limit is 1500 queued events for each partition.
```

Reply:
Fair point.   I hadn't thought of that.